sysvabi64/sysvabi64.rst

```diff
@@ -916,10 +923,652 @@ These follow the AAELF64_ base definition from Arm.
+a fixed virtual address. The underlying operating system maps shared
+objects at different virtual addresses, utilizing the fact that the
+code and data addressing within such a shared object is
+position-independent. The position independence relies on the
+principle that the offset between loadable program segments is fixed
+at static link time. This permits code to access data in another
+loadable program segment using a pc-relative offset that will not
+change at run time.
+
+Program segments for executables built with a fixed virtual address
+have to be mapped identically to the virtual addresses used to create
+the executable file.
+
+Position-independent executables are mapped in the same way as shared
+objects. They have an ELF ``e_type`` of ``ET_DYN``.
+
+Dynamic Section
+---------------
+
+The dynamic section entries allow information to be passed between the
+static linker and the dynamic linker. There are a number of entries
+that help locate various dynamic relocation tables and other aspects
+that are architecture independent.
+
+The AArch64 specific dynamic tags are defined in AAELF64_ Dynamic
+Section.
+
+Global Offset Table (GOT)
+-------------------------
+
+Position-independent code cannot, in general, contain absolute (fixed)
+virtual addresses. Global offset tables hold absolute addresses in
+private data, thus making the addresses available without compromising
+the position-independence and shareability of a program's text
+segment. A program references its global offset table using
+position-independent addressing and extracts the absolute values from
+it, thereby redirecting position-independent references to their
+actual locations.
+
+The global offset table (GOT) is created by the static linker in
+response to GOT generating relocations. See AAELF64_ Relocation
+operations for more information.
+
+AArch64 splits the global offset table (GOT) into two sections:
+
+ * ``.got.plt`` for code addresses accessed only from the Procedure Linkage
+   Table (PLT).
+
+ * ``.got`` all other addresses and offsets.
+
+Both the ``.got`` and ``.got.plt`` sections are aligned to a minimum
+of 8 bytes. They both contain 8-byte sized entries.
+
+If a position-independent program requires direct access to the
+absolute address of a symbol, that symbol will have a GOT
+entry. Because the executable and shared objects have their own
+independent GOTs, a symbol's address may appear in several tables. The
+dynamic linker processes all of the relocations in the ``.got``
+section before giving control to any code in the process image, thus
+ensuring the absolute addresses are available during execution. Code
+in the process that runs as part of a GNU indirect function resolver
+is an exception to this rule. See `GNU Indirect Functions`_ for the
+restrictions on what addresses can be accessed from a resolver
+function.
+
+By convention the ``.got`` section's first entry (number zero) is
```

Comment:
The  working about the first entry can probably be non-nominal (i.e. not required). lld does not set the entry to `_DYNAMIC`.

Reply:
I've tried a rewrite that essentially says that if you want to link an older version of glibc then you'll need to reserve the first entry for `_DYNAMIC`.